### ISSUE TITLE
Mouse actions are not properly implemented for absolute X/Y coordinate data

### ISSUE THREAD

**Issue by MarioSavu** [thewh1teagle/zero-hid#21]
The current list of supported mouse actions (button clicks, button release & scroll) have not been adapted to properly work with mouse absolute {X,Y} coordinate data.
The issue is that, unlike with relative {X,Y} coordinates, where leaving {0,0} for the X,Y coordinates in the various non movement functions would leave the cursor in place, if the same is applied for absolute the newly supported absolute coordinates, then the cursor will always jump back to {0,0} and end up in the top left corner after clicks & scrolls.

I've seen this happen with clicks myself, and the logs confirmed that this was the issue, as clicking would generate HID write logs with the coordinates having {0,0}. I expect scroll to have the same issue if it were tested based on reading the implementation.

In order to be able to utilize the same functions with the absolute {X,Y} mouse report descriptors, every report data that gets sent to the host needs to have the most recently used {X,Y} coordinates passed instead of {0,0}. And every move(), in case it's a __move_absolute(), needs to save that data so it can be used by click()s and scroll()s.
The implementation needs to be differentiated further in-between these 2 types of mouse report descriptors compared to the current implementation.

**Comment by thewh1teagle** [comment]
Hey!
Thanks for letting know. I didn't knew about that when using absolute coordinates.
It sounds like we need to take care of positions in every movement in both relative and absolute positioning.
I hope that there's simpler way to do it and still support absolute coordinates. Are you sure that in general when hid device send absolute reports it always sends the positions along with move / scroll reports instead of 0,0?

**Comment by thewh1teagle** [comment]
You can use relative device and absolute at the same time to achieve the result you wanted.

See

https://github.com/thewh1teagle/zero-hid/blob/main/examples/mouse_absolute_and_relative.py

And 

https://github.com/thewh1teagle/zero-hid?tab=readme-ov-file#gotachas

**Comment by MarioSavu** [comment]
Thanks for the quick response, I saw you already handled this by adding that example and writing up a gotcha for this particular limitation, but just to answer your question & make sure everything is clear:
> Hey! Thanks for letting know. I didn't knew about that when using absolute coordinates. It sounds like we need to take care of positions in every movement in both relative and absolute positioning. I hope that there's simpler way to do it and still support absolute coordinates. Are you sure that in general when hid device send absolute reports it always sends the positions along with move / scroll reports instead of 0,0?
At least from what I saw the library does and from what I vaguely know from the [HID spec](https://www.usb.org/sites/default/files/hid1_11.pdf), whenever a user of your library calls a function to change something (move the cursor, click or release a certain mouse button, move a scroll wheel), this results and has to result in a full report being sent. By full report I mean (at least for the current report descriptors that I see [here](https://github.com/thewh1teagle/zero-hid/blob/b4a75be959897a106ccae56ea1042fc33bdb7631/usb_gadget/init_usb_gadget) ) a 7 byte data packet containing: 1 byte to reflect the binary state of the 8 button + 2 bytes for X axis position (rel or abs), + 2 bytes for the Y axis + 1 byte for the vertical wheel rel position + 1 byte for the horizontal wheel. 
With this in mind, calling one of the intended functions to move the cursor, left click, and so on, should obviously change at least something from those 7 bytes, but the implementation shouldn't always be padding every other byte in that whole 7 byte report with zeroes. It's fine for relative data as a zero in that case will not change anything, but for absolute data like the button states, or in case of the absolute mouse the X & Y coordinates, it's not correct, as far as I can tell.

I saw you handled this for the button states through the usage of "self.buttons_state", so those don't seem to be handled okay, and in my mind it was just this X/Y axis in case of mouse absolute, so that one could potentially use just that implementation without having to even touch the mouse relative parts, and potentially even remove that HID device completely (but if someone does want that, they can just use their own library/implementation/fork this/whatever).

Anyway, you have the raw() function and anyone could use that instead of all this, and implement everything as they wished. 
The one worry I have there is this might have a bug: in the raw() function, you don't actually pass the input button_state into __send_mouse_event, instead you pass self.button_state. See [here](https://github.com/thewh1teagle/zero-hid/blob/main/zero_hid/mouse.py#L87). If this is meant to be a raw send mouse event function, shouldn't you pass the function input button_state rather than your classes latest button_state that's assigned and updated in the other click functions?
